### PR TITLE
New Relic, Dynatrace, Kairos, Wavefront, Stackdriver and SignalFX metrics.

### PR DESCRIPTION
**Pull request by thiagolocatelli**
New metrics integration

- micrometer-registry-dynatrace
- micrometer-registry-kairos
- micrometer-registry-new-relic
- micrometer-registry-signalfx
- micrometer-registry-stackdriver
- micrometer-registry-wavefront

Also, following the suggestion from @graemerocher, a fix for issue #17

### CODE REVIEW

File: micrometer-registry-new-relic/src/main/java/io/micronaut/configuration/metrics/micrometer/newrelic/NewRelicMeterRegistryFactory.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    public NewRelicMeterRegistryFactory(NewRelicConfigurationProperties newRelicConfigurationProperties) {
+        this.newRelicConfig = newRelicConfigurationProperties;
+    }
+
+    /**
+     * Create a NewRelicMeterRegistry bean if global metrics are enables
+     * and the newrelic is enabled.  Will be true by default when this
+     * configuration is included in project.
+     *
+     * @return A StackdriverMeterRegistry
```
Comment (graemerocher) [inline micrometer-registry-new-relic/src/main/java/io/micronaut/configuration/metrics/micrometer/newrelic/NewRelicMeterRegistryFactory.java]: Should be NewRelic

File: micrometer-registry-signalfx/src/main/java/io/micronaut/configuration/metrics/micrometer/signalfx/SignalFxMeterRegistryFactory.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+     * Sets the underlying signalfx meter registry properties.
+     *
+     * @param signalfxConfigurationProperties signalfx properties
+     */
+    public SignalFxMeterRegistryFactory(SignalFxConfigurationProperties signalfxConfigurationProperties) {
+        this.signalFxConfig = signalfxConfigurationProperties;
+    }
+
+    /**
+     * Create a StackdriverMeterRegistry bean if global metrics are enables
```
Comment (graemerocher) [inline micrometer-registry-signalfx/src/main/java/io/micronaut/configuration/metrics/micrometer/signalfx/SignalFxMeterRegistryFactory.java]: Should be SignalFx

File: micrometer-registry-new-relic/src/main/java/io/micronaut/configuration/metrics/micrometer/newrelic/NewRelicMeterRegistryFactory.java (hunk 1 above)
Comment (thiagolocatelli) [reply micrometer-registry-new-relic/src/main/java/io/micronaut/configuration/metrics/micrometer/newrelic/NewRelicMeterRegistryFactory.java]: Fixed

File: micrometer-registry-signalfx/src/main/java/io/micronaut/configuration/metrics/micrometer/signalfx/SignalFxMeterRegistryFactory.java (hunk 2 above)
Comment (thiagolocatelli) [reply micrometer-registry-signalfx/src/main/java/io/micronaut/configuration/metrics/micrometer/signalfx/SignalFxMeterRegistryFactory.java]: Fixed

**Comment by thiagolocatelli** [comment]
@graemerocher resolved changes. Micrometer has more registries I did not include, if you want I can work on the remaining registries so both Micronaut and Micrometer are in sync.

**Comment by graemerocher** [comment]
Sounds good!

**Comment by thiagolocatelli** [comment]
@graemerocher perfect, let me build them so they make it to 1.2.0. Would you mind taking a look at https://github.com/micronaut-projects/micronaut-core/issues/904 too? I want to get that piece of work done before core-1.2.0 goes out. Thank you, sir.

**Comment by ilopmar** [comment]
@thiagolocatelli could you please also create features for all these new implementations in the micronaut-profile repo?
You can use micronaut-prometheus (or any other existing ones) as an example: https://github.com/micronaut-projects/micronaut-profiles/tree/master/base/features/micrometer-prometheus
Thanks!

**Comment by thiagolocatelli** [comment]
@ilopmar I can definitely do that. I am working on the missing metrics and then will create feature for all of them, does this sound like a good plan?

**Comment by ilopmar** [comment]
@thiagolocatelli Excellent! :100:

**Comment by thiagolocatelli** [comment]
@ilopmar I am glad I can help, I wish I could do more.

**Pull request merged**